packages/blade/src/components/Dropdown/useDropdown.ts

```diff
@@ -0,0 +1,359 @@
+   */
+  const onTriggerBlur = (): void => {
+    if (rest.hasFooterAction) {
+      // When Footer has action buttons, we ignore the blur (by setting shouldIgnoreBlur to true in onTriggerKeyDown)
+      // And we remove the active item (by setting it to -1) so that we can shift focus on action buttons
+      setActiveIndex(-1);
+    }
+
+    if (shouldIgnoreBlur) {
+      setShouldIgnoreBlur(false);
+      return;
+    }
+
+    if (isOpen) {
+      if (selectionType !== 'multiple') {
+        selectOption(activeIndex);
+      }
+      setIsOpen(false);
+    }
+  };
+
+  /**
+   * Function that we call when we want to move focus from one option to other
+   */
+  const onOptionChange = (actionType: SelectActionsType, index?: number): void => {
+    const max = options.length - 1;
+    const newIndex = index ?? activeIndex;
+    setActiveIndex(getUpdatedIndex(newIndex, max, actionType));
+    const optionValues = options.map((option) => option.value);
+    ensureScrollVisiblity(newIndex, rest.actionListRef.current, optionValues);
+  };
+
+  /**
+   * Click handler when user clicks on any particular option.
+   *
+   * It
+   * - changes the option focus
+   * - selects that option
+   * - moves focus to combobox
+   */
+  const onOptionClick = (
+    e: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLInputElement>,
+    index: number,
+  ): void => {
+    const actionType = getActionFromKey(e, isOpen);
+    if (typeof actionType === 'number') {
+      onOptionChange(actionType, index);
+    }
+    selectOption(index);
+    if (!isReactNative()) {
+      rest.triggererRef.current?.focus();
+    }
+  };
+
+  /**
+   * Function we call to handle the typeahead.
+   *
+   * It takes a letter, stores that letter in searchString (and clears it after timeout) to maintain a word
+   *
+   * Then searches for that word in options and moves focus there.
+   */
+  const onComboType = (letter: string, actionType: SelectActionsType): void => {
+    // open the listbox if it is closed
+    setIsOpen(true);
+
+    if (typeof searchTimeout === 'number') {
+      window.clearTimeout(searchTimeout);
+    }
+
+    searchTimeout = window.setTimeout(() => {
+      searchString = '';
+    }, 500);
+    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+    searchString = searchString + letter;
+    const optionTitles = options.map((option) => option.title);
+    const searchIndex = getIndexByLetter(optionTitles, searchString, activeIndex + 1);
+
+    // if a match was found, go to it
+    if (searchIndex >= 0) {
+      onOptionChange(actionType, searchIndex);
+    }
+    // if no matches, clear the timeout and search string
+    else {
+      window.clearTimeout(searchTimeout);
+      searchString = '';
+    }
+  };
+
+  /**
+   * Keydown event of combobox. Handles most of the keyboard accessibility of dropdown
+   */
+  const onTriggerKeydown = (e: { event: React.KeyboardEvent<HTMLInputElement> }): void => {
```

Comment:
Why is this argument like `e.event` can we just pass in `event` directly?

Reply:
So on Select we're using `onKeyDown={onTriggerKeydown}` because our `BaseInput` by default passes `{ event }` object.

I am keeping things in that format only so that we can pass events like we need. E.g. in normal HTML button it will become `onKeyDown={(e) => onTriggerKeydown({ event: e.nativeEvent })}`